Form/Fields/RadioButton.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using DUCK.Utils;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace DUCK.Form.Fields
+{
+	[RequireComponent(typeof(Button))]
+	public class RadioButton : MonoBehaviour
+	{
+		public string Id
+		{
+			get { return id; }
+		}
+
+		[SerializeField]
+		private string id;
+
+		public event Action OnSelected;
+
+		public virtual void SetSelected(bool selected)
+		{
+		}
+
+		protected void InvokeOnSelected()
```

Comment:
I read this function from the subclass `CustomRadiobutton` it was confusing, it read like a boolean `invokeOnSelected = true` as if it would invoke when selected.
What you are looking for is the more standard `NotifySelected` - you are notifying the `RadioButton` it was selected then he can choose to do as he wishes (in this case dispatch event)

Reply:
Agreed